lib/honeycomb-beeline.rb

```diff
@@ -25,7 +25,7 @@ class << self
     extend Forwardable
     attr_reader :client
 
-    def_delegators :@client, :libhoney, :start_span, :add_field,
+    def_delegators :client, :libhoney, :start_span, :add_field,
```

Comment:
This is sort of a subtle change. By going through `:client` instead of `:@client`, this allows us to use RSpec to stub `Honeycomb.client` like

```ruby
allow(Honeycomb).to receive(:client).and_return(...)
```

And then `Honeycomb.start_span` & company will go through the stubbed client.

This is important because there's no way (short of `instance_variable_set`) to reset the client after a `Honeycomb.configure`. Thus, a `Honeycomb.configure` in some stray `before` block could leak into subsequent tests, which might break things depending on the order they're run. It's probably worth going through the rest of the suite and cleaning up the few stray uses of `Honeycomb.configure`.